crates/rpc/rpc-eth-api/src/helpers/state.rs

```diff
@@ -134,25 +134,26 @@ pub trait EthState: LoadState + SpawnBlocking {
         &self,
         address: Address,
         block_id: BlockId,
-    ) -> impl Future<Output = Result<Account, Self::Error>> + Send {
+    ) -> impl Future<Output = Result<Option<Account>, Self::Error>> + Send {
         self.spawn_blocking_io(move |this| {
             let state = this.state_at_block_id(block_id)?;
 
-            let account = state
-                .basic_account(address)
-                .map_err(Self::Error::from_eth_err)?
-                .unwrap_or_default();
-            let balance = account.balance;
-            let nonce = account.nonce;
-            let code_hash = account.bytecode_hash.unwrap_or(KECCAK_EMPTY);
+            let account = state.basic_account(address).map_err(Self::Error::from_eth_err)?;
+            if let Some(account) = account {
```

Comment:
styl nit:

let use `let Some() else {return Ok(None)}` here to get rid of the scope